plugins/auto-sizes/tests/test-auto-sizes.php

```diff
@@ -279,4 +288,103 @@ public function test_auto_sizes_update_content_img_tag( string $input, string $e
+	 */
+	public function test_auto_sizes_attribute_includes_valid_auto(): void {
+		// Test when 'auto' is the first item in the list.
+		$sizes_attr = 'auto, 100vw';
+		$this->assertTrue( auto_sizes_attribute_includes_valid_auto( $sizes_attr ) );
+
+		// Test when 'auto' is not the first item in the list.
+		$sizes_attr = '100vw, auto';
+		$this->assertFalse( auto_sizes_attribute_includes_valid_auto( $sizes_attr ) );
+	}
+
+	/**
+	 * Provides data for the auto_sizes_update_image_attributes test.
+	 *
+	 * @return array<int,mixed[]> The data for the test cases.
+	 */
+	public function data_provider_for_auto_sizes_update_image_attributes(): array {
+		return array(
+			array(
+				array(
+					'loading' => 'eager',
+					'sizes'   => '100vw',
+				),
+				array(
+					'loading' => 'eager',
+					'sizes'   => '100vw',
+				),
+			),
+
+			// Test when the image is not responsive.
+			array( array( 'loading' => 'lazy' ), array( 'loading' => 'lazy' ) ),
+
+			// Test when 'auto' already exists in the sizes attribute.
+			array(
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+			),
+
+			// Test when 'auto' needs to be added to the sizes attribute.
+			array(
+				array(
+					'loading' => 'lazy',
+					'sizes'   => '100vw',
+				),
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+			),
+		);
+	}
+
+	/**
+	 * @covers ::auto_sizes_update_image_attributes
+	 *
+	 * @dataProvider data_provider_for_auto_sizes_update_image_attributes
+	 *
+	 * @param array<string, mixed> $attr Attributes to be updated.
+	 * @param array<string, mixed> $expected Expected updated attributes.
+	 */
+	public function test_auto_sizes_update_image_attributes( array $attr, array $expected ): void {
+		$this->assertSame( $expected, auto_sizes_update_image_attributes( $attr ) );
+	}
+
+	/**
+	 * @covers ::auto_sizes_update_content_img_tag
+	 */
+	public function test_auto_sizes_update_content_img_tag_non_string_input(): void {
+		/*
+			These tests are separate from the data provider approach because the function
+			auto_sizes_update_content_img_tag() expects a string as an argument. Passing a non-string
+			value would cause a TypeError. These tests ensure that the function behaves as expected
+			when it receives non-string inputs.
+		*/
+		$this->assertSame( '', auto_sizes_update_content_img_tag( array() ) );
+	}
+
+	/**
+	 * @covers ::auto_sizes_update_image_attributes
+	 */
+	public function test_auto_sizes_update_image_attributes_with_null_input(): void {
+		/*
+			This test is separate because the main test method uses a data provider
+			that expects the $attr parameter to be an array. Passing null directly
+			would cause a TypeError due to the type hint. By testing null separately,
+			we ensure that the function can handle null inputs gracefully without
+			modifying the type hint in the main test method.
+		*/
```

Comment:
```suggestion
		/*
		 * This test is separate because the main test method uses a data provider
		 * that expects the $attr parameter to be an array. Passing null directly
		 * would cause a TypeError due to the type hint. By testing null separately,
		 * we ensure that the function can handle null inputs gracefully without
		 * modifying the type hint in the main test method.
		 */
```